features/unhandled_cpp_exception.feature

```diff
@@ -14,6 +14,7 @@ Feature: Thrown C++ exceptions are captured by Bugsnag
     And the event "severity" equals "error"
     And the event "unhandled" is true
     And the event "severityReason.type" equals "unhandledException"
+    And the event "threads.0.name" equals "BSG MAIN THREAD"
```

Comment:
Might be worth testing different values for the various cases to check that they are sensibly handled? for example, the following are valid:

```c
pthread_setname_np(thread, "");
pthread_setname_np(thread, "\nnext line?");
pthread_setname_np(current, "٥٦٧٨٩");
pthread_setname_np(current, "œ´¨ø“‘");
```